publishing.gradle

```diff
@@ -0,0 +1,165 @@
+                        name = subproject.name
+                        licenses = ["Apache-2.0"]
+                        desc = subproject.description
+                        websiteUrl = siteUrl
+                        vcsUrl = gitUrl
+                        dryRun = false
+                        publish = true
+                        override = true
+                        publicDownloadNumbers = false
+                        version {
+                            name = uport_sdk_version
+                            released = new Date()
+                        }
+                    }
+                }
+            }
+
+        }
+
+    }
+}
+
+//////////////////////////////////////////////////
+// helpers for loading publishing configuration //
+//////////////////////////////////////////////////
+
+def getBintrayApiKey() {
+    return getInfoFromPropOrVault("bintray_api_key")
+}
+
+def getBintrayUser() {
+    return getInfoFromPropOrVault("bintray_username")
+}
+
+def getInfoFromPropOrVault(String key) {
+
+    checkLocalPublishingProperties()
+    checkVaultEnvironment()
+
+    String info
+    if (project.publishProps?.get(key) != null) {
+        info = project.publishProps?.get(key)
+    } else if (project.hasProperty("IGNORE_VAULT")) {
+        info = ""
+    } else {
+        try {
+            info = project?.vault?.get("secret/android_sdk/$key")['data']['value']
+        } catch (Exception e) {
+            throw new InvalidUserDataException("There was an error while reading $key from vault.\n" +
+                    "Make sure to have secret/android_sdk/$key in the vault.\n" +
+                    "Run this task with -PIGNORE_VAULT to default to local configuration if available", e)
+        }
+    }
+    return info
+}
+
+def checkVaultEnvironment() {
+
+    if (project.hasProperty("IGNORE_VAULT")) {
+        //use only local data if available
+        return
+    }
+
+    File tokenFile = new File("${System.env.HOME}/.vault-token")
+
+    if (tokenFile.canRead()) {
+        if (System.env.VAULT_TOKEN == null || System.env.VAULT_TOKEN == "") {
+            project?.vault?.token = tokenFile.text
+        }
+    }
+
+    if (System.env.VAULT_ADDR == null || (System.env.VAULT_TOKEN == null && !tokenFile.canRead())) {
+        throw new InvalidUserDataException("Vault coordinates are not configured.\n" +
+                "VAULT_ADDR and VAULT_TOKEN need to be set as environment variables.\n" +
+                "Run this task with -PIGNORE_VAULT to skip secrets and build using local configuration only")
+    }
+}
+
+def checkLocalPublishingProperties() {
+    def propFile = file("${rootDir}/publish.properties")
+    if (propFile.canRead()) {
+
+        Properties publishProps = new Properties()
+        propFile.withInputStream {
+            publishProps.load(it)
+        }
+        project.ext.publishProps = publishProps
+    } else {
+        project.ext.publishProps = null
+    }
+}
```

Comment:
@mirceanis Is there a way to test this script?

Reply:
Yes, this exact script was used to publish v0.3.2 to jCenter as well